cmd/xl-v1-object.go

```diff
@@ -730,49 +730,54 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
 
 	isDir := hasSuffix(object, slashSeparator)
 
-	// If its a directory request, no need to read metadata.
 	if !isDir {
 		// Read metadata associated with the object from all disks.
 		metaArr, errs := readAllXLMetadata(xl.getDisks(), bucket, object)
-
 		// get Quorum for this object
 		_, writeQuorum, err = objectQuorumFromMeta(xl, metaArr, errs)
 		if err != nil {
 			return err
 		}
+		err = reduceWriteQuorumErrs(errs, objectOpIgnoredErrs, writeQuorum)
 	} else {
+		// WriteQuorum is defaulted to N/2 + 1 for directories
 		writeQuorum = len(xl.getDisks())/2 + 1
 	}
 
-	// Initialize list of errors.
-	var dErrs = make([]error, len(xl.getDisks()))
+	// No error, means proceed with deletion request
+	if err == nil {
```

Comment:
Can we return here if err is set so we can avoid a indent level in the next block ?

Reply:
good idea, done @vadmeste